src/components/ScanPatientPage/BarcodeScan.tsx

```diff
@@ -19,14 +25,21 @@ const BarcodeScan = ({ ccpID }: { ccpID: string }) => {
       if (selectedPatient.length > 0) {
         // Found patient
         const {
-          collectionPointId: { id: patientCCPId },
+          collectionPointId: {
+            id: patientCCPId,
+            eventId: { id: patientEventId },
+          },
           id,
         } = selectedPatient[0];
         // Redirect to patient profile
-        history.replace(`/patients/edit/${patientCCPId}/${id}`);
+        history.replace(
+          `/events/${patientEventId}/ccps/${patientCCPId}/patients/${id}`
+        );
       } else {
         // No existing patient
-        history.replace(`/patients/new/${ccpID}/${barcode}`);
+        history.replace(
+          `/events/${eventID}/ccps/${ccpID}/patients/new/${barcode}`
+        );
       }
     }
   }, [barcode]);
```

Comment:
added this `missing dependencies` warning to the other patient profile cleanup ticket instead